### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by ksigei**
### In this milestone, I was able to:
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student` and implemented the following:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classroom's students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. I implemented the following:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,17 +1,18 @@
 require_relative './nameable'
 require_relative './capitalize'
-require_relative './decorator'
+require_relative './rental'
 require_relative './trimmer'
 
 class Person < Nameable
   attr_accessor :name, :age, :id
-  attr_reader :parent_permission
+  attr_reader :parent_permission, :rentals
```
Comment (omar-muhamad) [inline person.rb]: - [ ] Person class suppose to have getters and setters for rentals not getters only, So Please fix that.

File: person.rb [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 
   def initialize(name = 'Unknown', age = 0, parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
     @id = rand(1..1000)
+    @rentals = []
     super()
```
Comment (omar-muhamad) [inline person.rb:16]: - [ ] Please add a method for adding rentals. Link to the [example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) in a previous lesson.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @ksigei,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] For student class Please add a method that serves as the 'belongs-to' side (a student belongs to a classroom). Link to the [article ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)with examples.

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb [hunk 3]
```diff
@@ -16,6 +16,10 @@ def initialize(name = 'Unknown', age = 0, parent_permission: true)
...
   end
 
+  def add_rental(rental)
+    @rentals << rental
+  end
+
```
Comment (DeliceLydia) [inline person.rb]: - [ ] *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association you can do:
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

Comment (DeliceLydia) [reply person.rb]: - [ ] This method `rental` has to be available in the `book` class as well and remember the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. you can do:
```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @ksigei, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (ksigei) [reply person.rb]: > 

i should use this in books.rb?

Comment (ksigei) [reply person.rb]: @DeliceLydia hello

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 

## Hi @ksigei 👋, 

Well done implementing all the required changes 👏 
Your project is now complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Highlights 🎯 
✔️ Good commit history
✔️ Correct GitFlow
✔️ Linter checks

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**